### PR TITLE
feat(replays): add toggles for block media and mask text

### LINKED ISSUE #61071: [Onboarding Sidebar] Dynamically configure loader or npm snippet wrt "Mask All Text y/n" and "Block All Media y/n"
The checkboxes (maskAllText etc) would apply to both loader and NPM. We'll have to write some extra code to perform the logic on which parts of the code snippets change

### PR DESCRIPTION
**Pull request by michellewzhang**
This PR adds two toggles, Mask All Text and Block All Media, next to the replay configuration step for both JS Loader and NPM instructions.

Also fixes a bug where some instructions were not rendering correctly based on the `?mode=` in the URL.

Closes https://github.com/getsentry/sentry/issues/61071

https://github.com/getsentry/sentry/assets/56095982/77082333-a20f-4235-bd5a-7a7827774c3e

### CODE REVIEW

File: static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx [hunk 1]
```diff
@@ -131,9 +138,36 @@ export function OnboardingLayout({
         </Header>
         <Divider withBottomMargin />
         <Steps>
-          {steps.map(step => (
-            <Step key={step.title ?? step.type} {...step} />
-          ))}
+          {steps.map(step =>
+            step.isOptional && step.isReplayConfigStep ? (
```
Comment (michellewzhang) [inline static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx:142]: if the step is optional, we have to shove the replay config toggle inside the step's `additionalInfo` to hide them

File: static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx [hunk 2]
```diff
@@ -131,9 +138,36 @@ export function OnboardingLayout({
...
+                      blockToggle={block}
+                      maskToggle={mask}
+                      onBlockToggle={() => setBlock(!block)}
+                      onMaskToggle={() => setMask(!mask)}
+                    />
+                  ),
+                }}
+              />
+            ) : (
+              <Fragment key={step.title ?? step.type}>
```
Comment (michellewzhang) [inline static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx:158]: otherwise the toggles sit under the config step

**Comment by c298lee** [comment]
I think it'll be a better experience to have the toggles above the code snippet, since the toggle moves off screen when turning it off depending on screen size

File: static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx (hunk 2 above)
Comment (michellewzhang) [reply static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx:158]: cc @c298lee

**Comment by ArthurKnaus** [comment]
@michellewzhang  I have the feeling we are reaching a point were we should separate replay specific features from the generic onboarding structure. Like implementing a separate `OnboardingLayout` component. Should not be a lot of work to do so wdyt?

File: static/app/components/onboarding/gettingStartedDoc/replayConfigToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          toggle={onBlockToggle}
+          size="lg"
+          isActive={blockToggle}
+        />
+      </SwitchItem>
+    </SwitchWrapper>
+  );
+}
+
+const SwitchItem = styled('div')`
```
Comment (ryan953) [inline static/app/components/onboarding/gettingStartedDoc/replayConfigToggle.tsx:42]: a11y!

```suggestion
const SwitchItem = styled('label')`
```

That html tag, paired with prop `<SwitchItem htmlFor="id-of-the-switch" />` would make the label clickable too. The `id=` prop of the `<Switch/>` doesn't need to be translated, or in sentence form either.

**Review by ryan953 — APPROVED** [review]

**Pull request merged**